core/domain/suggestion_services.py

```diff
@@ -348,3 +350,107 @@ def reject_suggestion(suggestion, reviewer_id, review_message):
     feedback_services.create_message(
         get_thread_id_from_suggestion_id(suggestion.suggestion_id), reviewer_id,
         feedback_models.STATUS_CHOICES_IGNORED, None, review_message)
+
+
+def get_user_contribution_scoring_from_model(userContributionScoringModel):
+    """Returns the UserContributionScoring domain object corresponding to the
+    UserContributionScoringModel
+
+    Args:
+        userContributionScoringModel: UserContributionScoringModel. The model
+            instance.
+
+    Returns:
+        UserContributionScoring. The corresponding domain object.
+    """
+    return user_domain.UserContributionScoring(
+        userContributionScoringModel.user_id,
+        userContributionScoringModel.score_category,
+        userContributionScoringModel.score)
+
+
+def get_all_scores_of_user(user_id):
+    """Gets all scores for a given user.
+
+    Args:
+        user_id: str. The id of the user.
+
+    Returns:
+        dict. A dict containing all the scores of the user. The keys of the dict
+            are the score categories and the values are the scores.
+    """
+    scores = {}
+    for model in (
+            user_models.UserContributionScoringModel.get_all_scores_of_user(
+                user_id)):
+        scores[model.score_category] = model.score
+
+    return scores
+
+
+def check_user_can_review_in_category(user_id, score_category):
+    """Checks if user can review suggestions in category score_category.
+    If the user has score above the minimum required score, then the user is
+    allowed to review.
+
+    Args:
+        user_id: str. The id of the user.
+        score_category: str. The category to check the user's score.
+
+    Returns:
+        bool. Whether the user can review suggestions under category
+            score_category
+    """
+    score = (
+        user_models.UserContributionScoringModel.get_score_of_user_for_category(
+            user_id, score_category))
+    if not score:
+        return False
+    return score >= feconf.MINIMUM_SCORE_REQUIRED_TO_REVIEW
+
+
+def get_all_user_ids_who_are_allowed_to_review(score_category):
```

Comment:
Technically, shouldn't this also include the editors/creator?

Reply:
Hmmm. For now I think we'll keep it without. When we do the rotational assignment, I'll edit this function. I thought I'll do it in a different way, and this function would just return the list of users to review as per score.